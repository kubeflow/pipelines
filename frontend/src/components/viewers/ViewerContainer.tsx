/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { ComponentType } from 'react';
import ConfusionMatrix from './ConfusionMatrix';
import HTMLViewer from './HTMLViewer';
import MarkdownViewer from './MarkdownViewer';
import PagedTable from './PagedTable';
import ROCCurve from './ROCCurve';
import TensorboardViewer from './Tensorboard';
import { PlotType, ViewerConfig } from './Viewer';
import VisualizationCreator from './VisualizationCreator';

export const componentMap: Record<PlotType, ComponentType<any>> = {
  [PlotType.CONFUSION_MATRIX]: ConfusionMatrix,
  [PlotType.MARKDOWN]: MarkdownViewer,
  [PlotType.ROC]: ROCCurve,
  [PlotType.TABLE]: PagedTable,
  [PlotType.TENSORBOARD]: TensorboardViewer,
  [PlotType.VISUALIZATION_CREATOR]: VisualizationCreator,
  [PlotType.WEB_APP]: HTMLViewer,
};

interface ViewerContainerProps {
  configs: ViewerConfig[];
  maxDimension?: number;
}

class ViewerContainer extends React.Component<ViewerContainerProps> {
  public render(): JSX.Element | null {
    const { configs, maxDimension } = this.props;
    if (!configs.length) {
      return null;
    }

    const Component = componentMap[configs[0].type];
    return <Component configs={configs as any} maxDimension={maxDimension} />;
  }
}

export default ViewerContainer;
