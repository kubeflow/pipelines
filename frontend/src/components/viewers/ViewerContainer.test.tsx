/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import { shallow } from 'enzyme';
import ViewerContainer from './ViewerContainer';
import { PlotType } from './Viewer';

describe('ViewerContainer', () => {
  it('does not break on empty configs', () => {
    const tree = shallow(<ViewerContainer configs={[]} />);
    expect(tree).toMatchSnapshot();
  });

  Object.keys(PlotType).map(type =>
    it('renders a viewer of type ' + type, () => {
      const tree = shallow(<ViewerContainer configs={[{ type: PlotType[type] }]} />);
      expect(tree).toMatchSnapshot();
    }),
  );
});
