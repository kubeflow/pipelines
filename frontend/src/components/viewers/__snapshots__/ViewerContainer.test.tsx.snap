// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewerContainer does not break on empty configs 1`] = `""`;

exports[`ViewerContainer renders a viewer of type CONFUSION_MATRIX 1`] = `
<ConfusionMatrix
  configs={
    Array [
      Object {
        "type": "confusion_matrix",
      },
    ]
  }
/>
`;

exports[`ViewerContainer renders a viewer of type MARKDOWN 1`] = `
<MarkdownViewer
  configs={
    Array [
      Object {
        "type": "markdown",
      },
    ]
  }
/>
`;

exports[`ViewerContainer renders a viewer of type ROC 1`] = `
<ROCCurve
  configs={
    Array [
      Object {
        "type": "roc",
      },
    ]
  }
/>
`;

exports[`ViewerContainer renders a viewer of type TABLE 1`] = `
<PagedTable
  configs={
    Array [
      Object {
        "type": "table",
      },
    ]
  }
/>
`;

exports[`ViewerContainer renders a viewer of type TENSORBOARD 1`] = `
<TensorboardViewer
  configs={
    Array [
      Object {
        "type": "tensorboard",
      },
    ]
  }
/>
`;

exports[`ViewerContainer renders a viewer of type VISUALIZATION_CREATOR 1`] = `
<VisualizationCreator
  configs={
    Array [
      Object {
        "type": "visualization-creator",
      },
    ]
  }
/>
`;

exports[`ViewerContainer renders a viewer of type WEB_APP 1`] = `
<HTMLViewer
  configs={
    Array [
      Object {
        "type": "web-app",
      },
    ]
  }
/>
`;
